token/js/src/instructions/associatedTokenAccount.ts

```diff
@@ -21,6 +22,57 @@ export function createAssociatedTokenAccountInstruction(
     mint: PublicKey,
     programId = TOKEN_PROGRAM_ID,
     associatedTokenProgramId = ASSOCIATED_TOKEN_PROGRAM_ID
+): TransactionInstruction {
+    return _createAssociatedTokenAccountInstruction(
+        payer,
+        associatedToken,
+        owner,
+        mint,
+        Buffer.alloc(0),
+        programId,
+        associatedTokenProgramId
+    );
+}
+
+/**
+ * Construct a CreateAssociatedTokenAccountIdempotent instruction
+ *
+ * @param payer                    Payer of the initialization fees
+ * @param associatedToken          New associated token account
+ * @param owner                    Owner of the new account
+ * @param mint                     Token mint account
+ * @param programId                SPL Token program account
+ * @param associatedTokenProgramId SPL Associated Token program account
+ *
+ * @return Instruction to add to a transaction
+ */
+export function createAssociatedTokenAccountIdempotentInstruction(
+    payer: PublicKey,
+    associatedToken: PublicKey,
+    owner: PublicKey,
+    mint: PublicKey,
+    programId = TOKEN_PROGRAM_ID,
+    associatedTokenProgramId = ASSOCIATED_TOKEN_PROGRAM_ID
+): TransactionInstruction {
+    return _createAssociatedTokenAccountInstruction(
+        payer,
+        associatedToken,
+        owner,
+        mint,
+        Buffer.from(new Uint8Array([1])),
+        programId,
+        associatedTokenProgramId
+    );
+}
+
+function _createAssociatedTokenAccountInstruction(
```

Comment:
nit: we don't use the `_functionName` convention in this repo for non-exported functions, so could you rename this to something like `buildAssociatedTokenAccountInstruction`? That way it'll match the rust implementation too